arista/utils/sonic_sfputil.py

```diff
@@ -98,32 +104,25 @@ def reset(self, port_num):
 
         def get_transceiver_change_event(self, timeout=0):
             xcvrs = inventory.getXcvrs()
-            epoll = select.epoll()
-            openFiles = []
             ret = {}
-            try:
-               # Clear the interrupt masks
-               for xcvr in xcvrs.values():
-                  intr = xcvr.getInterruptLine()
-                  if not intr:
-                     continue
-                  xcvr.getPresence()
-                  intr.clear()
-                  openFile = open(intr.getFile())
-                  openFiles.append((xcvr, openFile))
-                  epoll.register(openFile.fileno(), select.EPOLLIN)
-               pollRet = epoll.poll(timeout=timeout/1000. if timeout != 0 else -1)
-               if pollRet:
-                  pollRet = dict(pollRet)
-                  for xcvr, openFile in openFiles:
-                     if openFile.fileno() in pollRet:
-                        ret[str(xcvr.xcvrId)] = '1' if xcvr.getPresence() else '0'
-                  return True, ret
-            finally:
-               for _, openFile in openFiles:
-                  openFile.close()
-               epoll.close()
-
-            return True, {}
 
+            start_time = time.time()
+            timeout = timeout / float(1000) # convert msec to sec
+            while True:
+                for xcvr in xcvrs:
+                    presence = xcvr.getPresence()
+                    if self.xcvr_presence_map[xcvr.xcvrId] != presence:
+                        ret[str(xcvr.xcvrId)] = '1' if presence else '0'
+                        self.xcvr_presence_map[xcvr.xcvrId] = presence
+
+                if len(ret) != 0:
+                    return True, ret
+
+                if timeout != 0:
+                    elapsed_time = time.time() - start_time
+                    if elapsed_time >= timeout:
+                        return True, {}
+
+                # Poll for presence change every 1 second
+                time.sleep(1)
```

Comment:
I guess polling every second should be good enough.
The only downside of hardcoding 1 here is that timeouts that the caller might get a delta of 0/+1s compared to the provided timeout.
I don't believe this is a problem given how the xcvrd works but just want to make sure you're ok with this before I merge the PR